Explore dataset `World-Bank-Quiz.csv` dan jawab pertanyaan-pertanyaan yang diberikan.

# Memanggil library di Python

In [1]:
import pandas as pd

# Membaca file CSV sebagai data frame

In [2]:
world = pd.read_csv('World-Bank-Quiz.csv')

# Memahami data frame

In [3]:
world.shape

(117306, 67)

`world` terdiri dari 117306 baris dan 67 kolom.

Melihat preview `world`.

In [4]:
world.head()

Country Name Country Code  \
0  Africa Eastern and Southern          AFE   
1  Africa Eastern and Southern          AFE   
2  Africa Eastern and Southern          AFE   
3  Africa Eastern and Southern          AFE   
4  Africa Eastern and Southern          AFE   

                                      Indicator Name  Indicator Code  \
0  Adolescent fertility rate (births per 1,000 wo...     SP.ADO.TFRT   
1  Adults (ages 15+) and children (0-14 years) li...     SH.HIV.TOTL   
2  Adults (ages 15+) and children (ages 0-14) new...  SH.HIV.INCD.TL   
3                  Adults (ages 15+) living with HIV     SH.DYN.AIDS   
4        Adults (ages 15-49) newly infected with HIV     SH.HIV.INCD   

         1960        1961        1962       1963       1964        1965  ...  \
0  140.120535  140.594564  141.128109  141.32344  141.50816  141.668896  ...   
1         NaN         NaN         NaN        NaN        NaN         NaN  ...   
2         NaN         NaN         NaN        NaN        NaN         NaN  ...   
3         NaN         NaN         NaN        NaN        NaN         NaN  ...   
4         NaN         NaN         NaN        NaN        NaN         NaN  ...   

         2013        2014       2015       2016       2017       2018  \
0  103.391461  101.541782  99.719799  97.916601  96.146492  94.695223   
1         NaN         NaN        NaN        NaN        NaN        NaN   
2         NaN         NaN        NaN        NaN        NaN        NaN   
3         NaN         NaN        NaN        NaN        NaN        NaN   
4         NaN         NaN        NaN        NaN        NaN        NaN   

        2019  2020  2021  Unnamed: 66  
0  93.231944   NaN   NaN          NaN  
1        NaN   NaN   NaN          NaN  
2        NaN   NaN   NaN          NaN  
3        NaN   NaN   NaN          NaN  
4        NaN   NaN   NaN          NaN  

[5 rows x 67 columns]

Menampilkan informasi dari `world`.

In [5]:
world.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117306 entries, 0 to 117305
Data columns (total 67 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    117306 non-null  object 
 1   Country Code    117306 non-null  object 
 2   Indicator Name  117306 non-null  object 
 3   Indicator Code  117306 non-null  object 
 4   1960            37753 non-null   float64
 5   1961            38305 non-null   float64
 6   1962            38880 non-null   float64
 7   1963            38440 non-null   float64
 8   1964            38490 non-null   float64
 9   1965            38651 non-null   float64
 10  1966            38595 non-null   float64
 11  1967            39119 non-null   float64
 12  1968            38709 non-null   float64
 13  1969            38792 non-null   float64
 14  1970            40158 non-null   float64
 15  1971            40895 non-null   float64
 16  1972            41437 non-null   float64
 17  1973      

Meninjau duplicate data dari `world`.

In [6]:
world.duplicated().sum()

0

Meninjau null values dari `world`.

In [7]:
world.isna().sum()

Country Name           0
Country Code           0
Indicator Name         0
Indicator Code         0
1960               79553
                   ...  
2018               45923
2019               50240
2020               66339
2021              116253
Unnamed: 66       117306
Length: 67, dtype: int64

In [8]:
world.isna().sum().head(35)

Country Name          0
Country Code          0
Indicator Name        0
Indicator Code        0
1960              79553
1961              79001
1962              78426
1963              78866
1964              78816
1965              78655
1966              78711
1967              78187
1968              78597
1969              78514
1970              77148
1971              76411
1972              75869
1973              76318
1974              76278
1975              75309
1976              75231
1977              74739
1978              75165
1979              75078
1980              74248
1981              73963
1982              73675
1983              74058
1984              73833
1985              73564
1986              73178
1987              72713
1988              73361
1989              73109
1990              65011
dtype: int64

In [9]:
world.isna().sum().tail(35)

1988            73361
1989            73109
1990            65011
1991            64576
1992            63899
1993            64461
1994            64505
1995            64213
1996            64017
1997            63847
1998            64069
1999            62989
2000            45542
2001            48588
2002            47849
2003            48045
2004            47532
2005            45919
2006            46528
2007            45902
2008            46849
2009            46333
2010            42033
2011            45390
2012            44957
2013            46198
2014            44877
2015            44256
2016            44403
2017            45350
2018            45923
2019            50240
2020            66339
2021           116253
Unnamed: 66    117306
dtype: int64

Melihat value pada kolom `Indicator Name`.

In [10]:
world['Indicator Name'].unique()

array(['Adolescent fertility rate (births per 1,000 women ages 15-19)',
       'Adults (ages 15+) and children (0-14 years) living with HIV',
       'Adults (ages 15+) and children (ages 0-14) newly infected with HIV',
       'Adults (ages 15+) living with HIV',
       'Adults (ages 15-49) newly infected with HIV',
       'Age at first marriage, female', 'Age at first marriage, male',
       'Age dependency ratio (% of working-age population)',
       'Age dependency ratio, old', 'Age dependency ratio, young',
       'Age population, age 00, female, interpolated',
       'Age population, age 00, male, interpolated',
       'Age population, age 01, female, interpolated',
       'Age population, age 01, male, interpolated',
       'Age population, age 02, female, interpolated',
       'Age population, age 02, male, interpolated',
       'Age population, age 03, female, interpolated',
       'Age population, age 03, male, interpolated',
       'Age population, age 04, female, interpolated

# Menjawab pertanyaan-pertanyaan yang diberikan

### 1. Which country has the lowest death caused by nutrition in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [11]:
print(world[world['Indicator Name'].str.contains('nutrition', case = False)]['Indicator Name'].unique())

['Cause of death, by communicable diseases and maternal, prenatal and nutrition conditions (% of total)']


In [12]:
(world[world['Indicator Name'].str.contains('nutrition')]
 .loc[:, ['Country Name', 'Indicator Name', '2019']].sort_values('2019').head(3))

Country Name                                     Indicator Name  \
85623  North Macedonia  Cause of death, by communicable diseases and m...   
79890       Montenegro  Cause of death, by communicable diseases and m...   
51225          Finland  Cause of death, by communicable diseases and m...   

           2019  
85623  1.283611  
79890  1.413876  
51225  1.424167

<b>Answer: North Macedonia</b>

### 2. Which country has the highest rate of diabetes prevalence in 2021?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [13]:
print(world[world['Indicator Name'].str.contains('diabetes', case = False)]['Indicator Name'].unique())

['Diabetes prevalence (% of population ages 20 to 79)'
 'Mortality from CVD, cancer, diabetes or CRD between exact ages 30 and 70 (%)'
 'Mortality from CVD, cancer, diabetes or CRD between exact ages 30 and 70, female (%)'
 'Mortality from CVD, cancer, diabetes or CRD between exact ages 30 and 70, male (%)']


In [14]:
(world[world['Indicator Name'].str.contains('Diabetes prevalence')]
 .loc[:, ['Country Name', 'Indicator Name', '2021']].sort_values('2021', ascending = False).head(3))

Country Name                                     Indicator Name  \
87420          Pakistan  Diabetes prevalence (% of population ages 20 t...   
52140  French Polynesia  Diabetes prevalence (% of population ages 20 t...   
68457            Kuwait  Diabetes prevalence (% of population ages 20 t...   

       2021  
87420  30.8  
52140  25.2  
68457  24.9

<b>Answer: Pakistan</b>

### 3. How much is the working age population of Mali in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [15]:
print(world[world['Indicator Name'].str.contains('working', case = False)]['Indicator Name'].unique())

['Age dependency ratio (% of working-age population)']


In [16]:
(world[world['Indicator Name'].str.contains('working')]
 .loc[world['Country Name'] == 'Mali', ['Country Name', 'Indicator Name', '2020']])

Country Name                                     Indicator Name  \
75418         Mali  Age dependency ratio (% of working-age populat...   

           2020  
75418  97.95885

<b>Answer: 97.95885%</b>

### 4. How many Indonesian adults (age 15+) are living with HIV in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [17]:
print(world[world['Indicator Name'].str.contains('hiv', case = False)]['Indicator Name'].unique())

['Adults (ages 15+) and children (0-14 years) living with HIV'
 'Adults (ages 15+) and children (ages 0-14) newly infected with HIV'
 'Adults (ages 15+) living with HIV'
 'Adults (ages 15-49) newly infected with HIV'
 'Antiretroviral therapy coverage (% of people living with HIV)'
 'Antiretroviral therapy coverage for PMTCT (% of pregnant women living with HIV)'
 'Children (0-14) living with HIV'
 'Children (ages 0-14) newly infected with HIV'
 'Children orphaned by HIV/AIDS'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-24, female (2 prevent ways and reject 3 misconceptions)'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-24, male (2 prevent ways and reject 3 misconceptions)'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-49, female (2 prevent ways and reject 3 misconceptions)'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-49, male (2 prevent ways and reject 3 misconceptions)'
 'Incidence of HIV, ages 15-24 (per 1,000 uninfected population ages 15-24)

In [18]:
(world.loc[(world['Country Name'] == 'Indonesia') &
           (world['Indicator Name'] == 'Adults (ages 15+) living with HIV'),
           ['Country Name', 'Indicator Name', '2020']])

Country Name                     Indicator Name      2020
61302    Indonesia  Adults (ages 15+) living with HIV  520000.0

<b>Answer: 520000</b>

### 5. Which country has the lowest rate of literacy in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [19]:
print(world[world['Indicator Name'].str.contains('literacy', case = False)]['Indicator Name'].unique())

['Literacy rate, adult female (% of females ages 15 and above)'
 'Literacy rate, adult male (% of males ages 15 and above)'
 'Literacy rate, adult total (% of people ages 15 and above)'
 'Literacy rate, youth male (% of males ages 15-24)'
 'Literacy rate, youth total (% of people ages 15-24)']


In [20]:
(world[world['Indicator Name'].str.contains('Literacy rate, adult total')]
 .loc[:, ['Country Name', 'Indicator Name', '2020']].sort_values('2020').head(3))

Country Name  \
75566                                    Mali   
12503                              Low income   
6770   Heavily indebted poor countries (HIPC)   

                                          Indicator Name       2020  
75566  Literacy rate, adult total (% of people ages 1...  30.761410  
12503  Literacy rate, adult total (% of people ages 1...  61.368900  
6770   Literacy rate, adult total (% of people ages 1...  63.050411

<b>Answer: Mali</b>

### 6. Which country has the highest total fertility rate in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [21]:
print(world[world['Indicator Name'].str.contains('fertility', case = False)]['Indicator Name'].unique())

['Adolescent fertility rate (births per 1,000 women ages 15-19)'
 'Fertility rate, total (births per woman)'
 'Wanted fertility rate (births per woman)']


In [22]:
(world[world['Indicator Name'].str.contains('Fertility rate, total')]
 .loc[:, ['Country Name', 'Indicator Name', '2019']].sort_values('2019', ascending = False).head(3))

Country Name                            Indicator Name   2019
84791             Niger  Fertility rate, total (births per woman)  6.824
99344           Somalia  Fertility rate, total (births per woman)  5.978
41132  Congo, Dem. Rep.  Fertility rate, total (births per woman)  5.819

<b>Answer: Niger</b>

### 7. Which country has the lowest rate of diabetes prevalence in 2021?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [23]:
print(world[world['Indicator Name'].str.contains('diabetes', case = False)]['Indicator Name'].unique())

['Diabetes prevalence (% of population ages 20 to 79)'
 'Mortality from CVD, cancer, diabetes or CRD between exact ages 30 and 70 (%)'
 'Mortality from CVD, cancer, diabetes or CRD between exact ages 30 and 70, female (%)'
 'Mortality from CVD, cancer, diabetes or CRD between exact ages 30 and 70, male (%)']


In [24]:
(world[world['Indicator Name'].str.contains('Diabetes prevalence')]
 .loc[:, ['Country Name', 'Indicator Name', '2021']].sort_values('2021').head(3))

Country Name                                     Indicator Name  2021
30531         Benin  Diabetes prevalence (% of population ages 20 t...   1.1
53022   Gambia, The  Diabetes prevalence (% of population ages 20 t...   1.9
116967     Zimbabwe  Diabetes prevalence (% of population ages 20 t...   2.1

<b>Answer: Benin</b>

### 8. Which country has the lowest prevalence of overweight adults in 2016?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [25]:
print(world[world['Indicator Name'].str.contains('overweight', case = False)]['Indicator Name'].unique())

['Prevalence of overweight (% of adults)'
 'Prevalence of overweight (% of children under 5)'
 'Prevalence of overweight (modeled estimate, % of children under 5)'
 'Prevalence of overweight, female (% of children under 5)'
 'Prevalence of overweight, female (% of female adults)'
 'Prevalence of overweight, male (% of children under 5)'
 'Prevalence of overweight, male (% of male adults)']


In [26]:
(world.loc[world['Indicator Name'] == 'Prevalence of overweight (% of adults)',
           ['Country Name', 'Indicator Name', '2016']].sort_values('2016').head(3))

Country Name                          Indicator Name  2016
115005      Vietnam  Prevalence of overweight (% of adults)  18.3
61203         India  Prevalence of overweight (% of adults)  19.7
28569    Bangladesh  Prevalence of overweight (% of adults)  20.0

<b>Answer: Vietnam</b>

### 9. How many Indonesian children (age <15) are living with HIV in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [27]:
print(world[world['Indicator Name'].str.contains('hiv', case = False)]['Indicator Name'].unique())

['Adults (ages 15+) and children (0-14 years) living with HIV'
 'Adults (ages 15+) and children (ages 0-14) newly infected with HIV'
 'Adults (ages 15+) living with HIV'
 'Adults (ages 15-49) newly infected with HIV'
 'Antiretroviral therapy coverage (% of people living with HIV)'
 'Antiretroviral therapy coverage for PMTCT (% of pregnant women living with HIV)'
 'Children (0-14) living with HIV'
 'Children (ages 0-14) newly infected with HIV'
 'Children orphaned by HIV/AIDS'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-24, female (2 prevent ways and reject 3 misconceptions)'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-24, male (2 prevent ways and reject 3 misconceptions)'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-49, female (2 prevent ways and reject 3 misconceptions)'
 'Comprehensive correct knowledge of HIV/AIDS, ages 15-49, male (2 prevent ways and reject 3 misconceptions)'
 'Incidence of HIV, ages 15-24 (per 1,000 uninfected population ages 15-24)

In [28]:
(world.loc[(world['Country Name'] == 'Indonesia') &
           (world['Indicator Name'] == 'Children (0-14) living with HIV'),
           ['Country Name', 'Indicator Name', '2020']])

Country Name                   Indicator Name     2020
61371    Indonesia  Children (0-14) living with HIV  18000.0

<b>Answer: 18000</b>

### 10. Which country has the highest probability of dying per 1000 among children ages 5-9 years in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [29]:
print(world[world['Indicator Name'].str.contains('dying', case = False)]['Indicator Name'].unique())

['Probability of dying among adolescents ages 10-14 years (per 1,000)'
 'Probability of dying among adolescents ages 10-19 years (per 1,000)'
 'Probability of dying among adolescents ages 15-19 years (per 1,000)'
 'Probability of dying among children ages 5-9 years (per 1,000)'
 'Probability of dying among youth ages 20-24 years (per 1,000)']


In [30]:
(world[world['Indicator Name'].str.contains('dying among children')]
 .loc[:, ['Country Name', 'Indicator Name', '2020']].sort_values('2020', ascending = False).head(3))

Country Name                                     Indicator Name  2020
99598       Somalia  Probability of dying among children ages 5-9 y...  16.1
96952  Sierra Leone  Probability of dying among children ages 5-9 y...  15.9
85045         Niger  Probability of dying among children ages 5-9 y...  15.7

<b>Answer: Somalia</b>

### 11. Which country has the lowest total fertility rate in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [31]:
print(world[world['Indicator Name'].str.contains('fertility', case = False)]['Indicator Name'].unique())

['Adolescent fertility rate (births per 1,000 women ages 15-19)'
 'Fertility rate, total (births per woman)'
 'Wanted fertility rate (births per woman)']


In [32]:
(world[world['Indicator Name'].str.contains('Fertility rate, total')]
 .loc[:, ['Country Name', 'Indicator Name', '2019']].sort_values('2019').head(3))

Country Name                            Indicator Name   2019
67592           Korea, Rep.  Fertility rate, total (births per woman)  0.918
91406           Puerto Rico  Fertility rate, total (births per woman)  1.035
59654  Hong Kong SAR, China  Fertility rate, total (births per woman)  1.051

<b>Answer: Korea, Rep.</b>

### 12. Which country has 98% of their pregnant women receiving prenatal care in 2017?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [33]:
print(world[world['Indicator Name'].str.contains('pregnant', case = False)]['Indicator Name'].unique())

['Antiretroviral therapy coverage for PMTCT (% of pregnant women living with HIV)'
 'Intermittent preventive treatment (IPT) of malaria in pregnancy (% of pregnant women)'
 'Pregnant women receiving prenatal care (%)'
 'Pregnant women receiving prenatal care of at least four visits (% of pregnant women)'
 'Prevalence of anemia among non-pregnant women (% of women ages 15-49)'
 'Prevalence of anemia among pregnant women (%)'
 'Teenage mothers (% of women ages 15-19 who have had children or are currently pregnant)']


In [34]:
(world.loc[(world['Indicator Name'] == 'Pregnant women receiving prenatal care (%)') & (world['2017'] == 98),
           ['Country Name', 'Indicator Name', '2017']])

Country Name                              Indicator Name  2017
106170     Tanzania  Pregnant women receiving prenatal care (%)  98.0

<b>Answer: Tanzania</b>

### 13. Which country has the highest prevalence of overweight adults in 2016?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [35]:
print(world[world['Indicator Name'].str.contains('overweight', case = False)]['Indicator Name'].unique())

['Prevalence of overweight (% of adults)'
 'Prevalence of overweight (% of children under 5)'
 'Prevalence of overweight (modeled estimate, % of children under 5)'
 'Prevalence of overweight, female (% of children under 5)'
 'Prevalence of overweight, female (% of female adults)'
 'Prevalence of overweight, male (% of children under 5)'
 'Prevalence of overweight, male (% of male adults)']


In [36]:
(world.loc[world['Indicator Name'] == 'Prevalence of overweight (% of adults)',
           ['Country Name', 'Indicator Name', '2016']].sort_values('2016', ascending = False).head(3))

Country Name                          Indicator Name  2016
82371             Nauru  Prevalence of overweight (% of adults)  88.5
88104             Palau  Prevalence of overweight (% of adults)  85.1
76638  Marshall Islands  Prevalence of overweight (% of adults)  83.5

<b>Answer: Nauru</b>

### 14. Which country is more likely to have a measles outbreak due to the lowest rate of immunization in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [37]:
print(world[world['Indicator Name'].str.contains('measles', case = False)]['Indicator Name'].unique())

['Immunization, measles (% of children ages 12-23 months)'
 'Immunization, measles second dose (% of children by the nationally recommended age)']


In [38]:
(world.loc[world['Indicator Name'] == 'Immunization, measles (% of children ages 12-23 months)',
           ['Country Name', 'Indicator Name', '2019']].sort_values('2019').head(3))

Country Name                                     Indicator Name  \
88776  Papua New Guinea  Immunization, measles (% of children ages 12-2...   
38502              Chad  Immunization, measles (% of children ages 12-2...   
79956        Montenegro  Immunization, measles (% of children ages 12-2...   

       2019  
88776  37.0  
38502  41.0  
79956  42.0

<b>Answer: Papua New Guinea</b>

### 15. How much female age at first marriage in Slovenia, in 2018?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [39]:
print(world[world['Indicator Name'].str.contains('marriage', case = False)]['Indicator Name'].unique())

['Age at first marriage, female' 'Age at first marriage, male']


In [40]:
(world[world['Indicator Name'].str.contains('marriage, female')]
 .loc[world['Country Name'] == 'Slovenia', ['Country Name', 'Indicator Name', '2018']])

Country Name                 Indicator Name  2018
98348     Slovenia  Age at first marriage, female  34.0

<b>Answer: 34</b>

### 16. Which country has the highest rate of mortality by road traffic injury in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [41]:
print(world[world['Indicator Name'].str.contains('traffic', case = False)]['Indicator Name'].unique())

['Mortality caused by road traffic injury (per 100,000 people)'
 'Mortality caused by road traffic injury, female (per 100,000 female population)'
 'Mortality caused by road traffic injury, male (per 100,000 male population)']


In [42]:
(world.loc[world['Indicator Name'] == 'Mortality caused by road traffic injury (per 100,000 people)',
           ['Country Name', 'Indicator Name', '2019']].sort_values('2019', ascending = False).head(3))

Country Name                                     Indicator Name  \
46468   Dominican Republic  Mortality caused by road traffic injury (per 1...   
117028            Zimbabwe  Mortality caused by road traffic injury (per 1...   
114382       Venezuela, RB  Mortality caused by road traffic injury (per 1...   

        2019  
46468   64.6  
117028  41.2  
114382  39.0

<b>Answer: Dominican Republic</b>

### 17. Which country has 67% of their pregnant women receiving prenatal care in 2018?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [43]:
print(world[world['Indicator Name'].str.contains('pregnant', case = False)]['Indicator Name'].unique())

['Antiretroviral therapy coverage for PMTCT (% of pregnant women living with HIV)'
 'Intermittent preventive treatment (IPT) of malaria in pregnancy (% of pregnant women)'
 'Pregnant women receiving prenatal care (%)'
 'Pregnant women receiving prenatal care of at least four visits (% of pregnant women)'
 'Prevalence of anemia among non-pregnant women (% of women ages 15-49)'
 'Prevalence of anemia among pregnant women (%)'
 'Teenage mothers (% of women ages 15-19 who have had children or are currently pregnant)']


In [44]:
(world.loc[(world['Indicator Name'] == 'Pregnant women receiving prenatal care (%)') &
           (world['2018'] == 67), ['Country Name', 'Indicator Name', '2018']])

Country Name                              Indicator Name  2018
85443      Nigeria  Pregnant women receiving prenatal care (%)  67.0

<b>Answer: Nigeria</b>

### 18. How much is the prevalence of overweight adults in the United States, in 2016?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [45]:
print(world[world['Indicator Name'].str.contains('overweight', case = False)]['Indicator Name'].unique())

['Prevalence of overweight (% of adults)'
 'Prevalence of overweight (% of children under 5)'
 'Prevalence of overweight (modeled estimate, % of children under 5)'
 'Prevalence of overweight, female (% of children under 5)'
 'Prevalence of overweight, female (% of female adults)'
 'Prevalence of overweight, male (% of children under 5)'
 'Prevalence of overweight, male (% of male adults)']


In [46]:
(world.loc[(world['Indicator Name'] == 'Prevalence of overweight (% of adults)') &
           (world['Country Name'] == 'United States'), ['Country Name', 'Indicator Name', '2016']])

Country Name                          Indicator Name  2016
112800  United States  Prevalence of overweight (% of adults)  67.9

<b>Answer: 67.9%</b>

### 19. Which country is more likely to have TBC due to the lowest rate of immunization in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [47]:
print(world[world['Indicator Name'].str.contains('immunization', case = False)]['Indicator Name'].unique())

['Immunization, BCG (% of one-year-old children)'
 'Immunization, DPT (% of children ages 12-23 months)'
 'Immunization, HepB3 (% of one-year-old children)'
 'Immunization, Hib3 (% of children ages 12-23 months)'
 'Immunization, measles (% of children ages 12-23 months)'
 'Immunization, measles second dose (% of children by the nationally recommended age)'
 'Immunization, Pol3 (% of one-year-old children)']


In [48]:
(world[world['Indicator Name'].str.contains('Immunization, BCG')]
 .loc[:, ['Country Name', 'Indicator Name', '2019']].sort_values('2019').head(3))

Country Name                                  Indicator Name  2019
104207            Sweden  Immunization, BCG (% of one-year-old children)  25.0
99356            Somalia  Immunization, BCG (% of one-year-old children)  37.0
88772   Papua New Guinea  Immunization, BCG (% of one-year-old children)  52.0

<b>Answer: Sweden</b>

### 20. Which country has the average age of female at first marriage lower than 20 years old in 2016?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [49]:
print(world[world['Indicator Name'].str.contains('marriage', case = False)]['Indicator Name'].unique())

['Age at first marriage, female' 'Age at first marriage, male']


In [50]:
(world[world['Indicator Name'].str.contains('marriage, female')]
 .loc[world['2016'] < 20, ['Country Name', 'Indicator Name', '2016']])

Country Name                 Indicator Name  2016
74093       Malawi  Age at first marriage, female  19.9

<b>Answer: Malawi</b>

### 21. Which country has the highest lowest rate of mortality by road traffic injury in 2019?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [51]:
print(world[world['Indicator Name'].str.contains('traffic', case = False)]['Indicator Name'].unique())

['Mortality caused by road traffic injury (per 100,000 people)'
 'Mortality caused by road traffic injury, female (per 100,000 female population)'
 'Mortality caused by road traffic injury, male (per 100,000 male population)']


In [52]:
(world.loc[world['Indicator Name'] == 'Mortality caused by road traffic injury (per 100,000 people)',
           ['Country Name', 'Indicator Name', '2019']].sort_values('2019').head(3))

Country Name  \
24418    Antigua and Barbuda   
78220  Micronesia, Fed. Sts.   
75133               Maldives   

                                          Indicator Name  2019  
24418  Mortality caused by road traffic injury (per 1...   0.0  
78220  Mortality caused by road traffic injury (per 1...   0.2  
75133  Mortality caused by road traffic injury (per 1...   1.6

<b>Answer: Antigua and Barbuda</b>

### 22. How much is the working age population of Qatar in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [53]:
print(world[world['Indicator Name'].str.contains('working', case = False)]['Indicator Name'].unique())

['Age dependency ratio (% of working-age population)']


In [54]:
(world[world['Indicator Name'].str.contains('working')]
 .loc[world['Country Name'] == 'Qatar', ['Country Name', 'Indicator Name', '2020']])

Country Name                                     Indicator Name  \
91735        Qatar  Age dependency ratio (% of working-age populat...   

            2020  
91735  18.103287

<b>Answer: 18.103287%</b>

### 23. How much is the prevalence of overweight adults in Indonesia, in 2016?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [55]:
print(world[world['Indicator Name'].str.contains('overweight', case = False)]['Indicator Name'].unique())

['Prevalence of overweight (% of adults)'
 'Prevalence of overweight (% of children under 5)'
 'Prevalence of overweight (modeled estimate, % of children under 5)'
 'Prevalence of overweight, female (% of children under 5)'
 'Prevalence of overweight, female (% of female adults)'
 'Prevalence of overweight, male (% of children under 5)'
 'Prevalence of overweight, male (% of male adults)']


In [56]:
(world.loc[(world['Indicator Name'] == 'Prevalence of overweight (% of adults)') &
           (world['Country Name'] == 'Indonesia'), ['Country Name', 'Indicator Name', '2016']])

Country Name                          Indicator Name  2016
61644    Indonesia  Prevalence of overweight (% of adults)  28.2

<b>Answer: 28.2%</b>

### 24. Which country has the highest rate of literacy in 2020?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [57]:
print(world[world['Indicator Name'].str.contains('literacy', case = False)]['Indicator Name'].unique())

['Literacy rate, adult female (% of females ages 15 and above)'
 'Literacy rate, adult male (% of males ages 15 and above)'
 'Literacy rate, adult total (% of people ages 15 and above)'
 'Literacy rate, youth male (% of males ages 15-24)'
 'Literacy rate, youth total (% of people ages 15-24)']


In [58]:
(world[world['Indicator Name'].str.contains('Literacy rate, adult total')]
 .loc[:, ['Country Name', 'Indicator Name', '2020']].sort_values('2020', ascending = False).head(3))

Country Name  \
25292                                        Armenia   
79535                                       Mongolia   
5006   Europe & Central Asia (excluding high income)   

                                          Indicator Name       2020  
25292  Literacy rate, adult total (% of people ages 1...  99.788612  
79535  Literacy rate, adult total (% of people ages 1...  99.183273  
5006   Literacy rate, adult total (% of people ages 1...  99.077469

<b>Answer: Armenia</b>

### 25. How much is the average female age at first marriage in Indonesia, in 2017?

Meninjau `Indicator Name` sesuai keyword dari pertanyaan.

In [59]:
print(world[world['Indicator Name'].str.contains('marriage', case = False)]['Indicator Name'].unique())

['Age at first marriage, female' 'Age at first marriage, male']


In [60]:
(world[world['Indicator Name'].str.contains('marriage, female')]
 .loc[world['Country Name'] == 'Indonesia', ['Country Name', 'Indicator Name', '2017']])

Country Name                 Indicator Name  2017
61304    Indonesia  Age at first marriage, female  22.4

<b>Answer: 22.4</b>